### ISSUE TITLE
Propagate exit code from block process to top-level process

### ISSUE THREAD

**Issue by vsergeev** [vsergeev/luaradio#76]
When a child block process exits due to a runtime error, the error is not propagated to the top-level process. The exit information is available from `waitpid()` in `CompositeBlock:_reap()`, and could be returned to `CompositeBlock:wait()`, and then returned to the user, so the user can decide how to handle the erroneous exit.

**Comment by cdgraff** [comment]
Hi, @vsergeev I'll appreciate this improvement, are giving a lot of pain to manage restarts and failures without that.

Thanks!

**Comment by vsergeev** [comment]
Hi @cdgraff, I have the change working locally. I just need to clean up some things and update the C API. I'll try to get a new release out this week.

**Comment by vsergeev** [comment]
Hi @cdgraff, I've just released v0.11.0 with this improvement. CompositeBlock `wait()` and `stop()` now return a boolean indicating successful flowgraph exit. Let me know if that works for you and if you run into any other issues.

**Comment by cdgraff** [comment]
thanks! look good now!

```
11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:3] Starting process

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output FrequencyTranslatorBlock.out to input LowpassFilterBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output LowpassFilterBlock.out to input DownsamplerBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Aliased composite input TunerBlock.in to block input FrequencyTranslatorBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Aliased composite output TunerBlock.out to block output DownsamplerBlock.out

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output RtlSdrSource.out to input TunerBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output TunerBlock.out to input FrequencyDiscriminatorBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output FrequencyDiscriminatorBlock.out to input LowpassFilterBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output LowpassFilterBlock.out to input FMDeemphasisFilterBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output FMDeemphasisFilterBlock.out to input DownsamplerBlock.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Connected output DownsamplerBlock.out to input RealFileSink.in

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Flow graph:

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	RtlSdrSource [1102500 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [ComplexFloat32] -> {FrequencyTranslatorBlock.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	FrequencyTranslatorBlock [1102500 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [ComplexFloat32] <- {RtlSdrSource.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [ComplexFloat32] -> {LowpassFilterBlock.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	LowpassFilterBlock [1102500 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [ComplexFloat32] <- {FrequencyTranslatorBlock.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [ComplexFloat32] -> {DownsamplerBlock.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	DownsamplerBlock [220500 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [ComplexFloat32] <- {LowpassFilterBlock.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [ComplexFloat32] -> {FrequencyDiscriminatorBlock.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	FrequencyDiscriminatorBlock [220500 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [ComplexFloat32] <- {DownsamplerBlock.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [Float32] -> {LowpassFilterBlock.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	LowpassFilterBlock [220500 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [Float32] <- {FrequencyDiscriminatorBlock.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [Float32] -> {FMDeemphasisFilterBlock.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	FMDeemphasisFilterBlock [220500 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [Float32] <- {LowpassFilterBlock.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [Float32] -> {DownsamplerBlock.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	DownsamplerBlock [44100 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [Float32] <- {FMDeemphasisFilterBlock.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .out   [Float32] -> {RealFileSink.in}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	RealFileSink [44100 Hz]

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock]	    .in    [Float32] <- {DownsamplerBlock.out}

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Parent pid 1209

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block RtlSdrSource pid 1210

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block FrequencyTranslatorBlock pid 1211

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block RealFileSink pid 1219

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block DownsamplerBlock pid 1218

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] usb_claim_interface error -6

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block FMDeemphasisFilterBlock pid 1217

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block FrequencyDiscriminatorBlock pid 1215

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [RtlSdrSource] Block runtime error: [string "radio.blocks.sources.rtlsdr"]:0: rtlsdr_open(): -6

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] stack traceback:

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[C]: in function 'error'

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[string "radio.blocks.sources.rtlsdr"]: in function 'initialize_rtlsdr'

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[string "radio.blocks.sources.rtlsdr"]: in function 'run'

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[string "radio.core.composite"]: in function <[string "radio.core.composite"]:0>

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[C]: in function 'xpcall'

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[string "radio.core.composite"]: in function 'start'

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[string "radio.core.composite"]: in function 'run'

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	rtlsdr_wbfm_mono.lua:27: in main chunk

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[C]: in function 'dofile'

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	/usr/local/bin/luaradio:122: in main chunk

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 	[C]: at 0x563af68dd1d0

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block LowpassFilterBlock pid 1216

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block DownsamplerBlock pid 1214

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Block LowpassFilterBlock pid 1212

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [FrequencyTranslatorBlock] Block pid 1211 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [CompositeBlock] Child exited. Shutting down...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [LowpassFilterBlock] Block pid 1212 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [DownsamplerBlock] Block pid 1214 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [FrequencyDiscriminatorBlock] Block pid 1215 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [LowpassFilterBlock] Block pid 1216 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [FMDeemphasisFilterBlock] Block pid 1217 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [DownsamplerBlock] Block pid 1218 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] [RealFileSink] Block pid 1219 terminating...

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:4] 

11.11.22 11:17:29 (-0300) <encoder-fm-2> 2022/11/11 14:17:29 [input.external.rawaudio_0:3] Process exited with code 0

```
